api/v1alpha1/instaslice_types.go

```diff
@@ -56,105 +33,161 @@ const (
+
+	// nodeResources represents the resource list of the node
+	// +kubebuilder:validation:Required
+	NodeResources corev1.ResourceList `json:"nodeResources"`
+}
+
+type Mig struct {
+	// placements specify vendor profile indexes and sizes
+	// +kubebuilder:validation:Required
+	Placements []Placement `json:"placements"`
+
+	// giprofileID provides the GPU instance ID of a profile
+	// +kubebuilder:validation:Required
+	GIProfileID int32 `json:"giProfileId"`
+
+	// ciProfileID provides the compute instance ID of a profile
+	// +kubebuilder:validation:Required
+	CIProfileID int32 `json:"ciProfileId"`
+
+	// ciEngProfileID provides the compute instance engineering ID of a profile
+	// +kubebuilder:validation:Required
+	CIEngProfileID int32 `json:"ciEngProfileId,omitempty"`
+}
+
+type Placement struct {
+	// size represents slots consumed by a profile on GPU
+	// +kubebuilder:validation:Required
+	Size int32 `json:"size"`
+
+	// start represents the starting index driven by size for a profile
+	// +kubebuilder:validation:Required
+	Start int32 `json:"start"`
 }
 
-// InstasliceSpec defines the desired state of Instaslice
 type InstasliceSpec struct {
-	// migGpuUuid represents uuid of the mig device created on the gpu
-	// +required
-	MigGPUUUID map[string]string `json:"migGpuUuid"`
-	// allocations represents allocation details of user workloads
-	// +optional
-	Allocations map[string]AllocationDetails `json:"allocations"`
-	// migplacement represents gpu instance, compute instance with placement for a profile
-	// +required
-	Migplacement []Mig `json:"migplacement"`
-	// cpuonnodeatboot represents total amount of cpu present on the node
-	// +required
-	CpuOnNodeAtBoot int64 `json:"cpuonnodeatboot"`
-	// memoryonnodeatboot represents total amount of memory present on the node
-	// +required
-	MemoryOnNodeAtBoot int64 `json:"memoryonnodeatboot"`
+	// podAllocationRequests specifies the allocation requests per pod
+	// +kubebuilder:validation:optional
+	PodAllocationRequests map[types.UID]AllocationRequest `json:"podAllocationRequests"`
+
+	// emulatorMode specifies whether the Instaslice is running in emulator mode
+	// +kubebuilder:validation:Optional
+	EmulatorMode bool `json:"emulatorMode"`
 }
 
-// InstasliceStatus defines the observed state of Instaslice
 type InstasliceStatus struct {
-	// processed represents state of the instaslice object after daemonset creation
-	// +required
-	Processed bool `json:"processed"`
+	// podAllocationResults specify the allocation results per pod
+	// +kubebuilder:validation:optional
+	PodAllocationResults map[types.UID]AllocationResult `json:"podAllocationResults"`
+
+	// nodeResources specifies the discovered resources of the node
+	// +kubebuilder:validation:optional
+	NodeResources DiscoveredNodeResources `json:"nodeResources"`
+
+	// conditions represent the observed state of the Instaslice
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
Can you add an example to the description of what the conditions will look like?